src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -4,11 +4,25 @@
 
 from __future__ import absolute_import, division, print_function
 
+import os
 import threading
 
 from cryptography.hazmat.bindings._openssl import ffi, lib
 
 
+@ffi.callback("int (*)(unsigned char *, int)", error=-1)
```

Comment:
@reaperhulk do we need to call `ERR_put_error` on error here, or will stuff work ok without it?